project/src/main/java/com/google/sps/servlets/RequestAdminServlet.java

```diff
@@ -52,5 +60,48 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
         datastore.put(requestAdminEntity);
         response.sendRedirect("/login.html");
+        }
+        else {
+        String pending_email = request.getParameter("email");
+        String action = request.getParameter("action");
+        Query query = new Query("RequestAdmin");
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        ArrayList<String> users = new ArrayList<String>();
+        for (Entity entity:results.asIterable()){
+            if (entity.getProperty("email").equals(pending_email)){
+                datastore.delete(entity.getKey());
+            }
+        }
+        if (action.equals("approve")) {
+            query = new Query("User");
+            results = datastore.prepare(query);
+            for (Entity entity:results.asIterable()){
+                if (entity.getProperty("email").equals(pending_email)){
+                    entity.setProperty("admin", true);
+                    datastore.put(entity);
+                }
+            }
+        }
+        }
+    }
+
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        Query query = new Query("RequestAdmin");
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        ArrayList<Object> users = new ArrayList<Object>();
+        for (Entity entity:results.asIterable()){
+            ArrayList<String> user = new ArrayList<String>();
+            if (entity.getProperty("status").equals("pending")){
+                user.add((String) entity.getProperty("name"));
```

Comment:
try `Pair` class instead of using `ArrayList`, something like
```
Pair<String, String> pair = new Pair<>((String) entity.getProperty("name"), (String) entity.getProperty("email"));
```

see detail:
https://www.baeldung.com/java-pairs

Reply:
I agree right now that might work, but if it needs to be extended to include any other information about the user, reworking the `Pair` class implementation would require more work than the current `ArrayList` one. I'm also not sure if the key-value paradigm is exactly indicative of the information I want to return. Happy to change it if you think that's necessary, however. Also happy to hear other thoughts!

Reply:
I see. `Pair<Object, Object> pair` should work. But this is more like FYI. Feel free to keep `ArrayList`. The purpose of the suggest is to have some `key` in the return value so people know WHAT is each field instead of putting all information into list, when you have too many items, you would need go back to check which is the one you want.